source/MaterialXTest/MaterialXRender/RenderUtil.cpp

```diff
@@ -290,40 +290,45 @@ bool ShaderRenderTester::validate(const mx::FilePathVec& testRootPaths, const mx
             outputPath.removeExtension();
             for (const auto& element : elements)
             {
-                mx::TypedElementPtr targetElement = element;
-                mx::OutputPtr output = targetElement->asA<mx::Output>();
-                mx::ShaderRefPtr shaderRef = targetElement->asA<mx::ShaderRef>();
-                mx::NodePtr outputNode = targetElement->asA<mx::Node>();
-                mx::NodeDefPtr nodeDef = nullptr;
+                std::vector<mx::TypedElementPtr> targetElements;
+                std::vector<mx::NodeDefPtr> nodeDefs;
+
+                mx::OutputPtr output = element->asA<mx::Output>();
+                mx::ShaderRefPtr shaderRef = element->asA<mx::ShaderRef>();
+                mx::NodePtr outputNode = element->asA<mx::Node>();
+
                 if (output)
                 {
                     outputNode = output->getConnectedNode();
                     // Handle connected upstream material nodes later on.
                     if (outputNode->getType() != mx::MATERIAL_TYPE_STRING)
                     {
-                        nodeDef = outputNode->getNodeDef();
+                        nodeDefs.push_back(outputNode->getNodeDef());
```

Comment:
As GLSL has no nodeDef, don't we want to skip adding it in in this case. While OSL would find a nodedef.